wordpress-mobile/WordPress-Android
# Revert to Old Site Creation flow

Fixes #9102. It turns out the new site creation flow was enabled prematurely. This PR fully reverts #9048 and partially reverts a minor change in #9057.

Here are the commits that reverts the changes in #9048 which exactly matches the commits in that PR: 7faa1d52125579aac3db8d0f4f39e32d728a9926, 6363a85784b930b186acfc8b4c17368a4c068832, 0fea3487125c756b65d561806b02456a1ab631bd, 2e97ffa6e4ebc01affedfce8f1c861af62f5835d, bdc40c3f13106be76e2d14b052adf9ada16490bb, 66d15c2f8cebd20947ce6c0790a8dcf988d1bec3

64c55361b5d8e07b5891fa9b3aba3f2d969be6fc reverts only the changes for `site_creation_form_screen.xml` from #9057 since it's used in the old site creation flow.

The only new commit in this PR is the 647bf57617c33e4e0a93ddcba66712e50737cce6 which adds a specific flag for the new site creation flow, so we no longer depend on the `DEBUG` flag which was only being used for easier development.

To test:
* Run this PR as is and make sure the old site creation flow shows up. Please proceed to creating a site to verify that it's working as it used to. Here is me going through it: https://cloudup.com/cukPgM6W5a1
* Change the newly added `NEW_SITE_CREATION_ENABLED` flag in `WordPress/build.gradle` to `true` and make sure the new site creation flow shows up. Please proceed to creating a site. Here is me going through it: https://cloudup.com/cCjjimlpbav

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`. _We have reverted the change we previously added to release notes._
